nautobot/core/api/views.py

```diff
@@ -224,42 +222,23 @@ def get_object(self):
         return obj
 
     def get_serializer(self, *args, **kwargs):
-
         # If a list of objects has been provided, initialize the serializer with many=True
         if isinstance(kwargs.get("data", {}), list):
             kwargs["many"] = True
 
         return super().get_serializer(*args, **kwargs)
 
-    def get_serializer_class(self):
-
-        # If using 'brief' mode, find and return the nested serializer for this model, if one exists
-        if self.brief:
-            self.logger.debug("Request is for 'brief' format; initializing nested serializer")
-            try:
-                serializer = get_serializer_for_model(self.queryset.model, prefix="Nested")
-                self.logger.debug(f"Using serializer {serializer}")
-                return serializer
-            except SerializerNotFound:
-                self.logger.debug(f"Nested serializer for {self.queryset.model} not found!")
-
-        # Fall back to the hard-coded serializer class
-        return self.serializer_class
-
-    def get_queryset(self):
-        # If using brief mode, clear all prefetches from the queryset and append only brief_prefetch_fields (if any)
-        if self.brief:
-            # v2 TODO(jathan): Replace prefetch_related with select_related
-            return super().get_queryset().prefetch_related(None).prefetch_related(*self.brief_prefetch_fields)
-
-        return super().get_queryset()
+    def get_serializer_context(self):
+        context = super().get_serializer_context()
+        depth = 0
+        try:
+            depth = int(self.request.query_params.get("depth", 0))
+        except ValueError:
+            pass
```

Comment:
When would we expect a ValueError here? Should we log a message if it happens?

Reply:
When the `?depth` query parameter passed in here is not an integer. and Yeah def